### PR TITLE
feat(meshmultizoneservice): update status

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Checklist prior to review

Status updater for MeshMultiZoneService. We collect ports and MeshService that matched.

xref https://github.com/kumahq/kuma/issues/10507

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/config/core/resources/apis/config.go [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
 type Config struct {
+	// List of enabled core resources
 	Enabled []string `json:"enabled" envconfig:"KUMA_CORE_RESOURCES_ENABLED" default:""`
+	// Status of core resources
+	Status ConfigStatus `json:"status"`
+}
+
+type ConfigStatus struct {
+	// How often we compute status of MeshMultiZoneService
+	MeshMultiZoneServiceInterval config_types.Duration `json:"meshMultiZoneServiceInterval" envconfig:"CORE_RESOURCES_STATUS_MESH_MULTI_ZONE_SERVICE_INTERVAL"`
```
Comment (Automaat) [inline pkg/config/core/resources/apis/config.go:18]: shouldn't `envconfig` start with `KUMA_CORE_...` as `KUMA_CORE_RESOURCES_ENABLED`?

File: pkg/core/resources/apis/meshmultizoneservice/status_updater_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		}()
+
+		Expect(samples.MeshDefaultBuilder().Create(resManager)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		close(stopCh)
+	})
+
+	It("should add mesh services and port to the status of multizone service", func() {
```
Comment (Automaat) [inline pkg/core/resources/apis/meshmultizoneservice/status_updater_test.go:49]: Should we also add test for updating mesh service or ports? There is some logic with sorts etc, and we are not checking any of this by starting from scratch and adding service

File: pkg/core/resources/apis/meshmultizoneservice/status_updater.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	ctx := user.Ctx(context.Background(), user.ControlPlane)
+
+	for {
+		select {
+		case <-ticker.C:
+			start := time.Now()
+			if err := s.updateStatus(ctx); err != nil {
+				s.logger.Error(err, "could not update status of mesh multizone services")
+			}
+			s.metric.Observe(float64(time.Since(start).Milliseconds()))
```
Comment (Automaat) [inline pkg/core/resources/apis/meshmultizoneservice/status_updater.go:68]: don't we want to have metric with fails? It is easier to monitor and add alerting on metric than on logs

File: pkg/config/core/resources/apis/config.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/config/core/resources/apis/config.go:18]: uhhh, it definitely should! thx

File: pkg/core/resources/apis/meshmultizoneservice/status_updater.go (hunk 3 above)
Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/meshmultizoneservice/status_updater.go:68]: Hm, we don't do it usually. We could but at this point maybe it make sense to do this as "generic measurable ticker component". Adding this metric and writing ticker component is already a boilerplate. Let me create an issue. I'd like not to add it to this PR

Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/meshmultizoneservice/status_updater.go:68]: https://github.com/kumahq/kuma/issues/10671

**Review by Automaat — APPROVED** [review]

**Pull request merged**